actions/insights/general_setup.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Text, Dict, List
+
+from rasa_sdk import Tracker, FormValidationAction
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.types import DomainDict
+
+from actions.actions import all_required_slots_are_set
+from actions.slot_match import FuzzySlotMatch, FuzzySlotMatchOption, resolve_slot_match
+from common import logging
+
+
+logger = logging.initialize_logging()
+
+product_kind = FuzzySlotMatch(
+    "insights_product_kind",
+    [
+        FuzzySlotMatchOption("ansible"),
+        FuzzySlotMatchOption("rhel", ["rhel", "redhat", "red-hat", "red hat"]),
+        FuzzySlotMatchOption("openshift", ["openshift", "open shift"]),
+    ],
+)
+
+
+class InsightsGeneralSetup(FormValidationAction):
+    def name(self) -> Text:
+        return "validate_form_setup_insights"
+
+    async def extract_insights_product_kind(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: DomainDict
+    ) -> Dict[Text, Any]:
+        requested_slot = tracker.get_slot("requested_slot")
+        user_input = tracker.latest_message["text"]
+
+        if requested_slot == "insights_product_kind":
```

Comment:
Why have both of these two checks do the same thing?

Reply:
Slot pre-detection doesn't work without the other conditional statement. So, you cannot really say "help me setup insights for openshift", and expect it to give you the openshift utterance straightaway. Since, we are doing the same thing under each of these statements, I compiled the conditions into one block with an `or`.